eth_client/test/eth_client_test.exs

```diff
@@ -1,4 +1,48 @@
 defmodule EthClientTest do
   use ExUnit.Case
   doctest EthClient
+  alias EthClient.Account
+  alias EthClient.Context
+
+  @bin "../contracts/src/bin/Storage.bin"
+  @abi "../contracts/src/bin/Storage.abi"
+
+  describe "Ethereum war tooling functions" do
+    test "Deploy" do
+      bin = @bin
+      abi = @abi
+
+      contract = EthClient.deploy(bin, abi)
+      assert contract == EthClient.Context.contract()
+    end
```

Comment:
I see that the deploy process is repeated here, so I suggest using a setup:

```elixir
  @bin "../contracts/src/bin/Storage.bin"
  @abi "../contracts/src/bin/Storage.abi"
  
  describe "deploy/2" do
    setup %{bin: bin, abi: abi} do
         contract = EthClient.deploy(bin, abi)
         {:ok, contract: contract}
    end
    # This is just to illustrate the use of tags, 
    # as they can ultimately be left directly in the setup function.
    @tag input: ${bin: @bin, abi: @abi}
    test "When the contract is successfully generated", %{contract: } do
      assert contract == EthClient.Context.contract()
    end
    @tag input: ${bin: @bin, abi: @wrong_abi}
    test "When the contract is wrong", %{contract: } do
      refute contract == EthClient.Context.contract()
    end
  end
```

###  Some points I would like to highlight here are:

* The describe section is best put to reference the function to test `deploy/2`.
* The test section, must contain the scenario to test the function (test to fail, test to pass).
* It is more important to put scenarios where the test fails and to know that we are handling them correctly than to put the happy path, at the end the most natural thing is to use the happy path and that is generally more visible (even without tests).
* In the example, you should note the use of `refute/1` function instead of not equal operator to be more explicit in the test. You can use `refute/2` if you want to eval `false` or `nil` results.

**References:**
* [ExUnit.Assertion](https://hexdocs.pm/ex_unit/1.12.3/ExUnit.Assertions.html#refute/2)

Reply:
I just modificated the tests according to the suggestions. Among the new ones there are some to tests exceptions